### PR TITLE
KAFKA-3006: standardize KafkaConsumer API to use Collection

### PR DESCRIPTION
**Pull request by hachikuji**

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java [hunk 1]
```diff
@@ -1241,34 +1238,34 @@ public void pause(TopicPartition... partitions) {
...
-     * {@link #poll(long)} will return records from these partitions if there are any to be fetched.
-     * If the partitions were not previously paused, this method is a no-op.
-     * @param partitions The partitions which should be resumed
+     * Get the set of partitions that were previously paused by a call to {@link #pause(Collection)}.
+     *
+     * @return The set of paused partitions
      */
     @Override
-    public void resume(TopicPartition... partitions) {
+    public Set<TopicPartition> paused() {
```
Comment (gwenshap) [inline clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java]: Collection?

File: clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java [hunk 2]
```diff
@@ -137,19 +138,19 @@
     public Map<String, List<PartitionInfo>> listTopics();
 
     /**
-     * @see KafkaConsumer#pause(TopicPartition...)
+     * @see KafkaConsumer#paused()
      */
-    public void pause(TopicPartition... partitions);
+    public Set<TopicPartition> paused();
```
Comment (gwenshap) [inline clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java:143]: We are leaving this as a Set?

Comment (gwenshap) [inline clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java:143]: Sorry, out of KIP scope... never mind.

Comment (hachikuji) [inline clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java:143]: Yeah, sorry. This got shuffled around during rebasing.

**Comment by gwenshap** [comment]
LGTM. Waiting for tests to finish before merging.

**Comment by gwenshap** [comment]
Passed on my machine (I'm impatient!)

File: clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java [hunk 3]
```diff
@@ -137,19 +138,19 @@
...
      */
-    public Set<TopicPartition> paused();
+    public void pause(Collection<TopicPartition> partitions);
 
     /**
-     * @see KafkaConsumer#resume(TopicPartition...)
+     * @see KafkaConsumer#resume(Collection)
      */
-    public void resume(TopicPartition... partitions);
+    public void resume(Collection<TopicPartition> partitions);
```
Comment (lpicanco) [inline clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java:153]: There are some reason to move from varargs to Collections? It's breaking a lot of projects.

**Pull request closed without merging**